packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -181,10 +181,249 @@ export interface LocalSecondaryIndexProps extends SecondaryIndexProps {
   readonly sortKey: Attribute;
 }
 
+/**
+ * The base interface for a DynamoDB table.
+ *
+ * @experimental
+ */
+export interface ITable extends IResource {
+  /**
+   * The name of the DynamoDB table.
+   *
+   * @attribute
+   */
+  readonly tableName: string;
+
+  /**
+   * The ARN of the DynamoDB table.
+   *
+   * @attribute
+   */
+  readonly tableArn: string;
+
+  /**
+   * The Stream ARN of the DynamoDB table.
+   *
+   * @attribute
+   */
+  readonly tableStreamArn: string;
+
+  /**
+   * The partition (hash) key of the Table.
+   *
+   * @attribute
+   */
+  readonly tablePartitionKey: Attribute;
```

Comment:
Those should not be available here. They should only be available on the concrete construct and not for imported constructs.

Reply:
Yeah, this is an interesting discussion. I guess making it required means we can't ever have a `Table.fromTableArn` / `Table.fromTableName` method. But maybe there are other properties, like `tableStreamArn`, that already preclude it? If so, maybe we should just bite the bullet, and make them required here.

@RomainMuller your thoughts on the topic?

Reply:
Thing is it's very hard to actually *use* a DynamoDB table without knowing what it's key schema is... And if we want to have constructs that are "generic" like a "DynamoDBTableVIewer", having the key schema available makes things a lot nicer and easier. You don't have to duplicate the information about column names & types, ...

Reply:
OK, I'm fine with having these in `ITable` then. @eladb ?

Reply:
No, I don't feel comfortable expanding that surface area, especially given we want to expose all the configuration props of all resources in their concrete class. `ITable` is essentially a wrapper to the `tableArn` (and any methods that can operate at that level). It's mostly needed for **cross-app** scenarios and quite rare.

Reply:
Alright then - so I'm discarding this PR.